provider/provider.go

```diff
@@ -30,6 +33,27 @@ var (
 	ErrNonJSONResponse = errors.New("non JSON response")
 
 	errOnRelayRequest = errors.New("error on relay request")
+
+	regexPatterns = []*regexp.Regexp{}
+
+	errorStatusMap = map[string]int{
+		"context deadline exceeded (Client.Timeout exceeded while awaiting headers)": 408, // Request Timeout
+		"connection reset by peer":            503, // Service Unavailable
+		"no such host":                        404, // Not Found
+		"network is unreachable":              503, // Service Unavailable
+		"connection refused":                  502, // Bad Gateway
+		"http: server closed idle connection": 499, // Client Closed Request (non-standard)
+		"tls: handshake failure":              525, // SSL Handshake Failed (Cloudflare specific, non-standard)
+		"i/o timeout":                         504, // Gateway Timeout
+		"bad gateway":                         502, // Bad Gateway
+		"service unavailable":                 503, // Service Unavailable
+		"gateway timeout":                     504, // Gateway Timeout
+	}
+)
+
+const (
+	defaultCode = 202
```

Comment:
1. NIT: Can you please add an explanation above `202` as to why this is the default status code? If feeling to lazy, just copy-paste a link to https://github.com/pokt-foundation/pocket-go/pull/93#discussion_r1586616825.
2. `s/defaultCode/defaultStatusCode/g` to match the name of `errorStatusMap` above

Reply:
Did it too